app/server/api-generator/src/main/java/io/syndesis/server/api/generator/openapi/OpenApiPropertyGenerator.java

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.syndesis.server.api.generator.openapi;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import io.apicurio.datamodels.core.models.Extension;
+import io.apicurio.datamodels.core.models.common.SecurityScheme;
+import io.apicurio.datamodels.openapi.models.OasDocument;
+import io.syndesis.common.model.connection.ConfigurationProperty;
+import io.syndesis.common.model.connection.ConnectorSettings;
+import io.syndesis.server.api.generator.openapi.util.OasModelHelper;
+import org.apache.commons.lang3.StringUtils;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.ofNullable;
+
+/**
+ * Set of property generators that may apply to the give OpenAPI specification. When applicable property is added as configuration
+ * property. Subclasses provide 2.x and 3.x OpenAPI specific implementations.
+ */
+@SuppressWarnings("PMD.GodClass")
```

Comment:
Perhaps we could refactor this (at some point, not necessarily now) and break it into several classes to make it non god class. Something like one class per property.